test/torchaudio_unittest/librosa_compatibility_test.py

```diff
@@ -244,51 +252,76 @@ def assert_compatibilities_spectral_centroid(self, sample_rate, n_fft, hop_lengt
-            'n_fft': 600,
-            'hop_length': 100,
-            'power': 2.0,
-            'n_mels': 128,
-            'n_mfcc': 20,
-            'sample_rate': 16000
-        }
-        self.assert_compatibilities(**kwargs)
-
-    # NOTE: Test passes offline, but fails on TravisCI (and CircleCI), see #372.
-    @unittest.skipIf('CI' in os.environ, 'Test is known to fail on CI')
-    def test_basics3(self):
-        kwargs = {
-            'n_fft': 200,
-            'hop_length': 50,
-            'power': 2.0,
-            'n_mels': 128,
-            'n_mfcc': 50,
-            'sample_rate': 24000
-        }
-        self.assert_compatibilities(**kwargs)
-
-    def test_basics4(self):
-        kwargs = {
-            'n_fft': 400,
-            'hop_length': 200,
-            'power': 3.0,
-            'n_mels': 128,
-            'n_mfcc': 40,
-            'sample_rate': 16000
+        params_list = [
+            {
+                'n_fft': 400,
+                'hop_length': 200,
+                'power': 2.0,
+                'n_mels': 128,
+                'n_mfcc': 40,
+            },
+            {
+                'n_fft': 600,
+                'hop_length': 100,
+                'power': 2.0,
+                'n_mels': 128,
+                'n_mfcc': 20,
+            },
+            {
+                'n_fft': 400,
+                'hop_length': 200,
+                'power': 3.0,
+                'n_mels': 128,
+                'n_mfcc': 40,
+            },
+            {
+                'n_fft': 200,
+                'hop_length': 50,
+                'power': 2.0,
+                'n_mels': 128,
+                'n_mfcc': 50,
+                'skip_CI': True,  # NOTE: Test passes offline, but fails on TravisCI (and CircleCI), see #372.
+            },
+        ]
+        params_list = [
+            {'norm': norm, **params}
+            for params in params_list
+            for norm in [None, 'slaney']
+        ]
+
+        common_utils.set_audio_backend('default')
+        path = common_utils.get_asset_path('sinewave.wav')
+        sound, sample_rate = common_utils.load_wav(path)
+        sound_librosa = sound.cpu().numpy().squeeze()  # (64000)
+        data_dict = {
+            'sound': sound,
+            'sample_rate': sample_rate,
+            'sound_librosa': sound_librosa,
         }
-        self.assert_compatibilities(**kwargs)
+
+        for test_name, test_fn in compat_tests.items():
+            # what variables does our test function take
+            varnames = test_fn.__code__.co_varnames
+            params_list_restricted = [
+                {k: v for k, v in params.items() if k in varnames}
+                for params in params_list
+            ]
+            data_dict_restricted = {k: v for k, v in data_dict.items() if k in varnames}
+            # restrict to unique parameter combinations
+            params_list_unique = [dict(uniq) for uniq in {tuple(sorted(d.items())) for d in params_list_restricted}]
```

Comment:
I would ideally like to implement this in a util module and just have util.unique_dicts(params_list_restricted) here, so that it's clearer what this is doing and a comment isn't needed. Do you have a util module where this could go?

Reply:
I honestly do no get what this par of the code is trying to achieve, but I think this is unnecessary simply if you use `parameterized.expand`.

Reply:
Ye it's fighting python, it would just be list(set(params_list_restricted)) to get the unique dicts except that dicts a mutable and so unhashable. It's not needed now.